core/src/signer.ts

```diff
@@ -0,0 +1,30 @@
+import { ethers } from 'ethers';
+import NETWORKS from './constants/NETWORKS';
+import getProvider from './provider';
+
+const signers: Record<string, ethers.Wallet> = {};
+
+const getSigner = function (network: string): ethers.Wallet | undefined {
+    if (!NETWORKS[network]) {
+        throw new Error(`The network "${network}" is not supported yet!`);
+    }
+
+    if (!signers[network]) {
+        return undefined;
+    }
```

Comment:
This should throw an error instead. You don't want to check `getSigner` against undefined every time you used it.